src/GNNGraphs/transform.jl

```diff
@@ -188,6 +188,60 @@ function remove_multi_edges(g::GNNGraph{<:COO_T}; aggr = +)
              g.ndata, edata, g.gdata)
 end
 
+"""
+    remove_nodes(g::GNNGraph, nodes_to_remove::Vector{Int})
+
+Remove specified nodes, and their associated edges, from a GNNGraph.
+
+# Arguments
+- `g`: The input graph from which nodes (and their edges) will be removed.
+- `nodes_to_remove`: Vector of node indices to be removed.
+
+# Returns
+A new GNNGraph with the specified nodes and all edges associated with these nodes removed. 
+
+# Example
+```julia
+using GraphNeuralNetworks
+
+g = GNNGraph([1, 1, 2, 2, 3], [2, 3, 1, 3, 1])
+
+# Remove nodes with indices 2 and 3, for example
+g_new = remove_nodes(g, [2, 3])
+
+# g_new now does not contain nodes 2 and 3, and any edges that were connected to these nodes.
+println(g_new)
+```
+"""
+function remove_nodes(g::GNNGraph{<:COO_T}, nodes_to_remove)
+    s, t = edge_index(g)
+    w = get_edge_weight(g)
+    edata = g.edata
+    ndata = g.ndata
+
+    edges_to_remove = findall(x -> x in nodes_to_remove, s) ∪ findall(x -> x in nodes_to_remove, t)
```

Comment:
this can be made more efficient by sorting `nodes_to_remove` beforehand and then using `searchsortedlast`.

Reply:
for using that wont we need to assume that s and t arrays are sorted so that we can `searchsortedlast` for the node in s and t? and we cant do that to maintain the relations between s and t. 

couldn't make that work thus used this

let me know if I'm wrong

Reply:
I meant for `nodes_to_remove` to be sorted and for `searchsortedlast` to replace the `in`

Reply:
done